src/goto-programs/goto_program_template.h

```diff
@@ -80,60 +79,129 @@ class goto_program_templatet
     return *this;
   }
 
-  /*! \brief Container for an instruction of the goto-program
-  */
+  /// This class represents a node in the control flow graph of the function.
```

Comment:
Agreed. Do you know why there are two very similar representations goto programs? One reason @peterschrammel told me is that using one of them induces non-determinism in the analysis. Is this the main reason to have the other?